src/probnum/diffeq/odefilter/init_routines/_autodiff.py

```diff
@@ -0,0 +1,247 @@
+"""Automatic-differentiation-based initialization routines."""
+
+import numpy as np
+
+from probnum import problems, randprocs, randvars
+
+from ._interface import InitializationRoutine
+
+
+class _AutoDiffBase(InitializationRoutine):
+    def __init__(self):
+        super().__init__(is_exact=True, requires_jax=True)
+        self._jax, self._jnp = self._import_jax()
+
+    def _import_jax(self):
+
+        # pylint: disable="import-outside-toplevel"
+        try:
+            import jax
+            import jax.numpy as jnp
+            from jax.config import config
+
+            config.update("jax_enable_x64", True)
+
+            return jax, jnp
+
+        except ImportError as err:
+            raise ImportError(self._import_jax_error_msg) from err
+
+    @property
+    def _import_jax_error_msg(self):
+        return (
+            "Cannot perform Jax-based initialization without the optional "
+            "dependencies jax and jaxlib. "
+            "Try installing them via `pip install jax jaxlib`."
+        )
+
+    def __call__(
+        self,
+        *,
+        ivp: problems.InitialValueProblem,
+        prior_process: randprocs.markov.MarkovProcess,
+    ) -> randvars.RandomVariable:
+
+        num_derivatives = prior_process.transition.num_derivatives
+
+        f, y0 = self._make_autonomous(ivp=ivp)
+
+        mean_matrix = self._compute_ode_derivatives(
+            f=f, y0=y0, num_derivatives=num_derivatives
+        )
+        mean = mean_matrix.reshape((-1,), order="F")
+        zeros = self._jnp.zeros((mean.shape[0], mean.shape[0]))
+        return randvars.Normal(
+            mean=np.asarray(mean),
+            cov=np.asarray(zeros),
+            cov_cholesky=np.asarray(zeros),
+        )
+
+    def _compute_ode_derivatives(self, *, f, y0, num_derivatives):
+        gen = self._F_generator(f=f, y0=y0)
+        mean_matrix = self._jnp.stack(
+            [next(gen)(y0)[:-1] for _ in range(num_derivatives + 1)]
+        )
+        return mean_matrix
+
+    def _make_autonomous(self, *, ivp):
+        y0_autonomous = self._jnp.concatenate([ivp.y0, self._jnp.array([ivp.t0])])
+
+        def f_autonomous(y):
+            x, t = y[:-1], y[-1]
+            fx = ivp.f(t, x)
+            return self._jnp.concatenate([fx, self._jnp.array([1.0])])
+
+        return f_autonomous, y0_autonomous
```

Comment:
One might argue that this functionality should be implemented in the `IVP` class, since it is not the job of the initializer to re-formalize initial value problems

Reply:
Hmmmm maybe... it is simply to simplify the calls to autodiff routines. Since this functionality can be isolated so strictly to this use case, isnt it easiest to parse if implemented here? What do you think?

Reply:
There are no changes made to the IVP. ``f`` and ``y0`` are wrapped into a format that is more convenient for the autodiff init-routine. It is purely internal, and at the moment, I cannot see any potential to reuse this functionality anywhere else. If such potential comes up, we can move it to a different place, but at the moment, I think it is well-placed here for this reason.
What do you think? :) 

In any case, I take from this the need to leave a little docstring in this function that explains precisely that rationale. (That is part of the most recent commit).